lib/ansible/galaxy/role.py

```diff
@@ -96,16 +96,17 @@ def metadata(self):
         Returns role metadata
         """
         if self._metadata is None:
-            meta_path = os.path.join(self.path, self.META_MAIN)
-            if os.path.isfile(meta_path):
-                try:
-                    f = open(meta_path, 'r')
-                    self._metadata = yaml.safe_load(f)
-                except:
-                    display.vvvvv("Unable to load metadata for %s" % self.name)
-                    return False
-                finally:
-                    f.close()
+            for meta_main in self.META_MAIN:
+                meta_path = os.path.join(self.path, meta_main)
```

Comment:
i would just add 'meta' here  and do 1 join vs 2 of em

Reply:
That would require to add another `join` on the line 273. I think it has better readability as it's now.